From 25f89bbd4e3722b771b93965d5f9827e27ca2614 Mon Sep 17 00:00:00 2001
From: gan <Gan Yuchou>
Date: Fri, 25 Mar 2016 21:24:38 +0800
Subject: [PATCH 9/9] MGS-1678 [#2269] memleak in GPU driver sysfs interface

Add missing .release callback in file_operations of vidmem_operations in order to release the allocated memory.

Date: Mar 18, 2016
Signed-off-by: Yuchou Gan <yuchou.gan@nxp.com>
(cherry picked from commit 667aa43c5c79dd4885dd71c5bc2296804d31ada0)
[zou:Original patch taken from
git.freescale.com/imx/fsl-arm-yocto-bsp.git -b
imx-3.14.52-1.1.0_ga]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 .../hal/os/linux/kernel/gc_hal_kernel_debugfs.c    |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_debugfs.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_debugfs.c
index 5a51730..fa54275 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_debugfs.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_debugfs.c
@@ -892,6 +892,7 @@ static const struct file_operations vidmem_operations = {
     .read = seq_read,
     .write = vidmem_write,
     .llseek = seq_lseek,
+    .release = single_release,
 } ;
 
 /*******************************************************************************
-- 
1.7.5.4

